src/Illuminate/Auth/EloquentUserProvider.php

```diff
@@ -117,7 +117,7 @@ public function retrieveByCredentials(array $credentials)
         $query = $this->newModelQuery();
 
         foreach ($credentials as $key => $value) {
-            if (Str::contains($key, 'password')) {
+            if (Str::contains($key, $query->getModel()->getPasswordName())) {
```

Comment:
There's a `array_key_exists('password', $credentials)` check just a few lines above this one.